src/reconcile/rec_child.c

```diff
@@ -94,7 +96,7 @@ __rec_child_deleted(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REF *ref, WT_C
      * Otherwise, we have to write a proxy record. If the delete state is not ready, then delete is
      * not visible as it is in prepared state.
      */
-    if (!__wt_page_del_active(session, ref, false))
+    if (!__wt_page_del_active(session, ref, !F_ISSET(txn, WT_TXN_HAS_SNAPSHOT)))
```

Comment:
Is this still necessary? Didn't @kommiharibabu change it recently so all eviction threads have snapshots?

Reply:
Yes, that may not be required.